compiler/fir/tree/src/org/jetbrains/kotlin/fir/types/FirTypeUtils.kt

```diff
@@ -95,13 +96,23 @@ fun List<FirAnnotationCall>.computeTypeAttributes(): ConeAttributes {
         when (type.lookupTag.classId) {
             CompilerConeAttributes.Exact.ANNOTATION_CLASS_ID -> attributes += CompilerConeAttributes.Exact
             CompilerConeAttributes.NoInfer.ANNOTATION_CLASS_ID -> attributes += CompilerConeAttributes.NoInfer
+            CompilerConeAttributes.EnhancedNullability.ANNOTATION_CLASS_ID -> attributes += CompilerConeAttributes.EnhancedNullability
             CompilerConeAttributes.ExtensionFunctionType.ANNOTATION_CLASS_ID -> attributes += CompilerConeAttributes.ExtensionFunctionType
             CompilerConeAttributes.UnsafeVariance.ANNOTATION_CLASS_ID -> attributes += CompilerConeAttributes.UnsafeVariance
+            in NOT_NULL_ANNOTATION_IDS -> attributes += CompilerConeAttributes.EnhancedNullability
         }
     }
     return ConeAttributes.create(attributes)
 }
 
+// Not to load [org/jetbrains/kotlin/load/java/JvmAnnotationNames.JETBRAINS_NOT_NULL_ANNOTATION] at :compiler.common.jvm
```

Comment:
I guess we prefer less dependencies? Not sure how to deal with other annotations in `NOT_NULL_ANNOTATIONS`, though.

Reply:
This looks quite alien here. I think we should introduce some JVM-specific `computeTypeAttributes` and call it from enhancement. Basic `fir:tree` should know nothing about these annotations.

Reply:
Ah, didn't think of that, and `fir:java` has a dependency on `compiler.common.jvm` ([link](https://github.com/JetBrains/kotlin/blob/master/compiler/fir/java/build.gradle.kts#L8)) of course. 🤦 

Added a JVM-specific version with an additional processor notion.